lib/Validation.groovy

```diff
@@ -0,0 +1,105 @@
+@Grab(group='org.everit.json', module='org.everit.json.schema', version='1.5.1')
```

Comment:
I wonder what happens if you try to run this on a fresh system without an Internet connection 🤔

Reply:
Good point ... that would only be relevant if someone wants to download the pipeline though, right? Otherwise, you need internet for lots of other things anyway.
But there must be some way of 'installing' it.

Reply:
Nextflow includes Groovy [JsonSlurper](https://groovy-lang.org/json.html). Could not be used instead of downloading `org.everit.json.schema` ?

Reply:
The idea was to use a Schema validator to avoid having to code the functionality ourselves - so the issue is not so much in parsing the JSON but in validating it against the Schema.

Reply:
Okay so groovy stores the dependencies by default in `~/.groovy/grape` 
So if you started the workflow once online, the dependencies are saved in it works offline as well.
It is also possible to directly install dependencies to `~/.groovy/grape` using `grape install`. The directory to install the dependencies to can also be changed.

If someone runs a pipeline offline, they will usually have to run `nf-core download` before, right? So we could simply add a function to that which installs the java dependencies with grape. Should be doable. I'm testing and collecting the necessary commands to get it running ... :man_technologist:

Reply:
Okay, all we need to do is to run these two commands when downloading the pipeline:

```
grape install org.json json [20201115]
grape install org.everit.json org.everit.json.schema [1.5.1]
```

Then it works offline, too.

Reply:
Ah ok sorry just saw this. Great, but won't work for 99% of people using `nf-core download` sadly, as this gets it offline _on the current system_. If it's not in the resulting tarball with the pipeline files then it won't be transferred to the offline cluster.

Reply:
> Nextflow includes Groovy [JsonSlurper](https://groovy-lang.org/json.html). Could not be used instead of downloading `org.everit.json.schema` ?

@pditommaso we're not using this library to load JSON, we're using it to validate _JSON Schema_ - one of these things that looks simple on the face of it and turns out to be really very complex..

Reply:
I see. Then, provided this lib self-contained (ie does not pull other libraries) you can put that jar in the project `lib/` directory and NF adds it to the class path.